### PR TITLE
BEP-5: Token Resources [IMPROVEMENT]

### PR DESCRIPTION
**Pull request by jpthor**
This BEP describes an improvement to the BEP-2 Token standard to add resources for token management on the Binance Chain.

Token Resources are managed by a URI that is specified as part of an issued token on Binance Chain.

### CODE REVIEW

File: BEP5.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    "logo": "https://token.io/logo.png",
+    "logo_large": "https://token.io/logo_large.png",
+  },
+  "other": {
+    "****": "****",
+  }
+}
+```
+
+### 5.3 JSON SCHEMA
```
Comment (huangsuyu) [inline BEP5.md]: This data structure is quite complicated and not easy for adopting future changes.

Comment (jpthor) [reply BEP5.md]: Hi @huangsuyu 

To clarify, the only addition to the protocol is the following field which needs to be added to the token and is less than 120 bytes:

```
	URI: "https://token.io/token_data.json"   <--- 120byte string
```

The URI can be changed/updated at any time. This makes it very easy for adopting future changes. 

The client (DEX/wallet/DAPP) then retrieves the json packet and displays the data. This data is off-chain and is *very* flexible. 

As a minimum standard, all that needs to be added are the following fields:

- description
- icon
- website

Everything else is optional but enhances the DEX/Wallet/DAPP, and can be updated at any time. 

To sum up:

On-chain: 120byte string
Off-chain queried by clients: json packet

Please let me know if you have more questions.

**Comment by yutianwu** [comment]
Actually these resources are rather centralized. What if one of those links is invalid or just changed. One purpose of these resources is to prevent scams. But if the token owner can decide which icon to use, it will be much easier for scammers.

The alternative solution is we can share these resources in a repo. So explorers can share those resources. The resources can be audited so it can also prevent scams.

**Comment by hiddentao** [comment]
Why not have the resource URI point to IPFS or something similar, so that it's content-addressable and thus not prone to that sort of scamming?

**Comment by darren-liu** [comment]
hi @jpthor,

I will suggest we give up BEP-5 here, as now there is a github.com/binance-chain/tokens-info repository to store all the information. Storing everything on the chain sounds an appealing idea, but there are benefits to store the information on github.

1. much more friendly accessible via all the existing git and github toolings, to users, designer, and managers of all levels
2. easy to change, explore and link, including all the history

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by iakisme** [comment]
@jpthor: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by brilliant-lx** [comment]
outdate, archive it.

**Comment by Rubens1969** [comment]
***@***.***

**Comment by blaaauw** [comment]
Hi

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
